File: src/main/java/com/example/fileservice/rest/ResponseEntity.java
package com.example.fileservice.rest;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
public class ResponseEntity<T> {

    @Setter
    private T data;

    @Setter
    private List<ErrorMessage> errors;

    private final int status;

    public ResponseEntity(int status) {
        this.status = status;
    }

    public ResponseEntity(T data, int status) {
        this(data, null, status);
    }

    public ResponseEntity(T data, List<ErrorMessage> errors, int status) {
        this.data = data;
        this.errors = errors;
        this.status = status;
    }
}

File: src/main/kotlin/com/example/fileservice/rest/ResponseEntity.kt
package com.hrblizz.fileapi.rest

class ResponseEntity<T> {
    var data: T? = null

    var errors: List<ErrorMessage>? = null

    var status: Int = 0
        private set

    constructor(status: Int) {
        this.status = status
    }

    constructor(data: T?, status: Int) : this(data, null, status) {}

    constructor(data: T?, errors: List<ErrorMessage>?, status: Int) {
        this.data = data
        this.errors = errors
        this.status = status
    }
}
